libuserhelper.cpp

```diff
@@ -388,3 +410,55 @@ QStringList LibUserHelper::groups(uint uid)
 
     return rv;
 }
+
+QString LibUserHelper::getUserUUID(uint uid) const
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
+    if (!context) {
+        qCWarning(lcSUM) << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return QString();
+    }
+
+    struct lu_ent *ent = lu_ent_new();
+
+    QString userUuid;
+    if (lu_user_lookup_id(context, uid, ent, &error)) {
+        QString gecos = lu_ent_get_first_string(ent, LU_GECOS);
+        const auto list = gecos.split(',');
+        if (list.size() > 1 && !list[1].isEmpty()) {
+            userUuid = list[1];
+        } else {
+            // try to generate user uuid
+            if (modifyUser(uid, gecos)) {
```

Comment:
I think this can't be `gecos` as it may contain `,` characters. It could be `list.count() >= 1 ? list[0] : QString()` I think.

Reply:
Yeah, hard to see with things without ut. I fix this.